src/App.tsx

```diff
@@ -1,15 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { ITaskList } from '@components/types';
+import { v4 as uuidv4 } from 'uuid';
 import Title from './components/Title';
 import AddItemForm from './components/AddItemForm';
 import TaskDashboard from './components/TaskDashboard';
 
 function App() {
+  const [taskList, setListTask] = useState<ITaskList[]>([]);
+
+  const handleAddTask = (text: string) => {
+    const newList: any = { ...taskList };
+
+    const uniqueId: string = uuidv4();
+
+    newList[uniqueId] = {
+      id: uniqueId,
+      textTask: text,
+      taskComplete: false,
+    };
+
+    setListTask(newList);
+  };
+
```

Comment:
Fiz algumas alterações no code como um todo e simplifiquei a função `handleAddTask` (poderia se chamar `addTask`). Perceba, como você já tem a interface `Task`, para se ter uma lista de tasks basta escrever `Task[]`. Com isto, também deletei o tipo ITaskList, pois ele representava um objeto, e não precisamos mais dele.

```suggestion
function App() {
  const [taskList, setListTask] = useState<Task[]>([]);

  const handleAddTask = (text: string) => {
    const newTask: Task = {
      id: uuidv4(),
      textTask: text,
      taskComplete: false,
    };

    setListTask([...taskList, newTask]);
  };

```